pkg/scheduler/framework/plugins/defaultassigner/algorithm.go

```diff
@@ -17,6 +17,51 @@ type weightList struct {
 	weights   []int64
 }
 
+// DynamicDivideReplicas will fill the target replicas of all feeds based on predictor result and feed finv.
+func DynamicDivideReplicas(selected *framework.TargetClusters, sub *appsapi.Subscription, clusters []*clusterapi.ManagedCluster, finv *appsapi.FeedInventory) error {
+	if selected.Replicas == nil {
+		selected.Replicas = make(map[string][]int32)
+	}
+	for _, feed := range sub.Spec.Feeds {
+		var order *appsapi.FeedOrder
+		for i := range finv.Spec.Feeds {
+			if feed == finv.Spec.Feeds[i].Feed {
+				order = &finv.Spec.Feeds[i]
+				break
+			}
+		}
+		if order != nil && order.DesiredReplicas != nil {
+			replicas := dymamicDivideReplicas(*order.DesiredReplicas, selected.Replicas[order.Name])
+			selected.Replicas[utils.GetFeedKey(order.Feed)] = replicas
+		} else {
+			selected.Replicas[utils.GetFeedKey(order.Feed)] = []int32{}
+		}
+	}
+	return nil
+}
+
+// dynamicDivideReplicas divides replicas by the MaxAvailableReplicas
+func dymamicDivideReplicas(desiredReplicas int32, maxAvailableReplicas []int32) []int32 {
```

Comment:
Seems like now this function only cares about disperse replicas in all candidate clusters. We need more strategies in the future so I have reserved the API before. It's OK, we may design this in next PR.

Reply:
If we have a new strategy, we can add a argument here and implement corresponding algorithm in next pr.